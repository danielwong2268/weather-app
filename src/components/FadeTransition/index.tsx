import React from 'react';
import { css } from 'emotion';
import { CSSTransition } from 'react-transition-group';

const fadeStyle = css`
  .fade-in-enter {
    transition: opacity 250ms ease-in;
    transform: translateY(10px);
    opacity: 0.01;
  }

  .fade-in-enter-active {
    opacity: 1;
    transform: translateY(0px);
    transition: all 250ms ease-out;
  }
`;

interface FadeProps {
  children: React.ReactNode;
  inProp: boolean;
}

const FadeTransition = ({
  children,
  inProp
}: FadeProps) => (
  <div className={fadeStyle}>
    <CSSTransition
      unmountOnExit
      in={inProp}
      timeout={{ enter: 250, exit: 250 }}
      classNames="fade-in"
    >
      <div>{children}</div>
    </CSSTransition>
  </div>
);

export default FadeTransition;