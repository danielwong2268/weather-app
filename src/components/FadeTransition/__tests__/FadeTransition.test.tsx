import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FadeTransition from '..';

describe('<FadeTransition />', () => {
  it('Renders with children', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<FadeTransition inProp>child</FadeTransition>);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});