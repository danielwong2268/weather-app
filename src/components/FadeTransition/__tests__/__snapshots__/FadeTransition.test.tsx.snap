// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FadeTransition /> Renders with children 1`] = `
<div
  className="css-3u43w9"
>
  <CSSTransition
    classNames="fade-in"
    in={true}
    timeout={
      Object {
        "enter": 250,
        "exit": 250,
      }
    }
    unmountOnExit={true}
  >
    <div>
      child
    </div>
  </CSSTransition>
</div>
`;
